libraries/libstratosphere/source/lm/detail/lm_detail_log_observer.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2018-2020 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <stratosphere.hpp>
+
+namespace ams::lm::detail {
+
+    namespace {
+
+        os::SdkMutex g_log_observer_lock;
+        u8 g_packet_transmitter_buffer[0x400];
+
+        bool LogPacketTransmitterFlushFunction(const void *log_data, size_t log_data_size) {
+            /* TODO: libnx bindings. */
+            return true;
+        }
+
+        constexpr const char DefaultInvalidString[] = "(Invalid UTF8 string)";
+
+    }
+
+    void LogManagerLogObserver(const diag::LogMetaData &log_metadata, const diag::LogBody &log_body, void *user_data) {
```

Comment:
This (and other code) creates lots of locals that are non-const and never used again. Better off with immediates.